openpype/hosts/max/plugins/publish/validate_camera_contents.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+from openpype.pipeline import PublishValidationError
+from pymxs import runtime as rt
+
+
+class ValidateCameraContent(pyblish.api.InstancePlugin):
+    """Validates Camera instance contents.
+
+    A Camera instance may only hold a SINGLE camera's transform
+    """
+
+    order = pyblish.api.ValidatorOrder
+    families = ["camera"]
+    hosts = ["max"]
+    label = "Camera Contents"
+    camera_type = ["$Free_Camera", "$Target_Camera",
+                   "$Physical_Camera", "$Target"]
+
+    def process(self, instance):
+        invalid = self.get_invalid(instance)
+        if invalid:
+            raise PublishValidationError("Camera instance must only include"
+                                         "camera (and camera target)")
+
+    def get_invalid(self, instance):
+        """
+        Get invalid nodes if the instance is not camera
+        """
+        invalid = list()
+        container = instance.data["instance_node"]
+        self.log.info("Validating look content for "
+                      "{}".format(container))
+
+        con = rt.getNodeByName(container)
+        selection_list = self.list_children(con)
+        validation_msg = list()
+        for sel in selection_list:
+            # to avoid Attribute Error from pymxs wrapper
+            sel_tmp = str(sel)
+            for cam in self.camera_type:
+                if sel_tmp.startswith(cam):
+                    validation_msg.append("Camera Found")
+                else:
+                    validation_msg.append("Camera Not Found")
+            if "Camera Found" not in validation_msg:
+                invalid.append(sel)
```

Comment:
I would simplify this:

```suggestion
        for sel in selection_list:
            # to avoid Attribute Error from pymxs wrapper
            sel_tmp = str(sel)
            found = False
            for cam in self.camera_type:
                if sel_tmp.startswith(cam):
                    found = True
                    break
            if not found:
                self.log.error("Camera not found")
                invalid.append(sel)
```

Reply:
I actually think check code-wise is odd to read.

We seem to be comparing the object type to see if it's a camera. 

I'd search on how to explicitly get the object type instead of assuming it's the start of the string for `str(object)` of a 3ds max object so that we could turn the whole validation .py file into something along these lines:

```python
# -*- coding: utf-8 -*-
import pyblish.api
from openpype.pipeline import PublishValidationError
from pymxs import runtime as rt


def is_valid_camera_node(node):
    obj_type = node.type()  # or whatever the valid method is for that
    return obj_type in {"$Free_Camera", "$Target_Camera", 
                        "$Physical_Camera", "$Target"}
    

class ValidateCameraContent(pyblish.api.InstancePlugin):
    """Validates Camera instance contents.

    A Camera instance may only hold a SINGLE camera's transform
    """

    order = pyblish.api.ValidatorOrder
    families = ["camera"]
    hosts = ["max"]
    label = "Camera Contents"

    def process(self, instance):
        invalid = self.get_invalid(instance)
        if invalid:
            raise PublishValidationError("Camera instance must only include"
                                         "camera (and camera target)")

    def get_invalid(self, instance):
        """Get invalid nodes if the instance is not camera"""
        
        container = rt.getNodeByName(instance.data["instance_node"])

        invalid = list()
        for child in list(container.Children):
            if not is_valid_camera_node(child):
                invalid.append(child)
                
        return invalid

```

Reply:
Had another quick peek over the 3ds Max `pymxs` docs. I think it should be something along these lines:

One of these:
```
rt.superClassOf(node) 
rt.classOf(node)
```

And then check it against the actual `rt` types:

```python
def is_valid_camera_node(node):
    obj_type = rt.classOf(node)
    return obj_type in {rt.Free_Camera, rt.Target_Camera,
                        rt.Physical_Camera, rt.Target}
```

Having never used 3ds Max I'm not sure that's anywhere remotely valid - but it seemed like that should be roughly it. :) But do consider it pseudocode!